File: app/src/main/java/com/bkjcb/rqapplication/check/fragment/ApplianceCheckItemDetailFragment.java
package com.bkjcb.rqapplication.check.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.ScrollView;
import android.widget.TextView;

import com.bkjcb.rqapplication.R;
import com.bkjcb.rqapplication.check.model.ApplianceCheckContentItem;
import com.bkjcb.rqapplication.check.model.ApplianceCheckResultItem;
import com.bkjcb.rqapplication.check.model.ApplianceCheckResultItem_;
import com.bkjcb.rqapplication.fragment.BaseLazyFragment;

/**
 * Created by DengShuai on 2020/1/7.
 * Description :
 */
public class ApplianceCheckItemDetailFragment extends BaseLazyFragment implements RadioGroup.OnCheckedChangeListener {

    private View view;
    private ApplianceCheckContentItem contentItem;
    private String uid;//项目id
    private ApplianceCheckResultItem checkResultItem;
    private EditText mItemRecord;
    private EditText mItemRemark;
    private RadioButton mCheckResultRadioOk;
    private RadioButton mCheckResultRadioFailure;
    private RadioGroup mCheckResultRadioGroup;
    private boolean type;
    private ScrollView mScrollView;

    public void setType(boolean type) {
        this.type = type;
    }

    public void setContentItem(ApplianceCheckContentItem contentItem) {
        this.contentItem = contentItem;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public static ApplianceCheckItemDetailFragment newInstances(ApplianceCheckContentItem contentItem, String id) {
        ApplianceCheckItemDetailFragment fragment = new ApplianceCheckItemDetailFragment();
        fragment.setContentItem(contentItem);
        fragment.setUid(id);
        return fragment;
    }

    public static ApplianceCheckItemDetailFragment newInstances(ApplianceCheckContentItem contentItem, String id, boolean type) {
        ApplianceCheckItemDetailFragment fragment = new ApplianceCheckItemDetailFragment();
        fragment.setContentItem(contentItem);
        fragment.setUid(id);
        fragment.setType(type);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (view == null) {
            view = inflater.inflate(R.layout.fragment_appliance_check_item_detail, null);
            initView(view);

        }

        return view;
    }

    @Override
    public void onStart() {
        super.onStart();
        checkResultItem = queryLocalItem();
        if (checkResultItem != null) {
            mItemRecord.setText(checkResultItem.content);
            mItemRemark.setText(checkResultItem.remark);
            if (!TextUtils.isEmpty(checkResultItem.ischeck)) {
                if ("1".equals(checkResultItem.ischeck)) {
                    mCheckResultRadioGroup.check(R.id.check_result_radio_ok);
                    changeTextColor(true);

                } else if ("0".equals(checkResultItem.ischeck)) {
                    mCheckResultRadioGroup.check(R.id.check_result_radio_failure);
                    changeTextColor(false);
                }
            }
        } else {
            checkResultItem = new ApplianceCheckResultItem(uid, contentItem.getGuid());
        }
    }


    private ApplianceCheckResultItem queryLocalItem() {
        return ApplianceCheckResultItem.getBox().query().equal(ApplianceCheckResultItem_.jianchaid, uid).and().equal(ApplianceCheckResultItem_.jianchaxiangid, contentItem.getGuid()).build().findFirst();
    }

    private void initView(View view) {
        TextView mItemType = (TextView) view.findViewById(R.id.item_type);
        TextView mItemContent = (TextView) view.findViewById(R.id.item_content);
        TextView mItemSection = (TextView) view.findViewById(R.id.item_section);
        mCheckResultRadioOk = (RadioButton) view.findViewById(R.id.check_result_radio_ok);
        mCheckResultRadioFailure = (RadioButton) view.findViewById(R.id.check_result_radio_failure);
        mCheckResultRadioGroup = (RadioGroup) view.findViewById(R.id.check_result_radioGroup);
        TextView mItemRecordTitle = (TextView) view.findViewById(R.id.item_record_title);
        mItemRecord = (EditText) view.findViewById(R.id.item_record);
        LinearLayout mItemRecord1 = (LinearLayout) view.findViewById(R.id.item_record1);
        TextView mItemRemarkTitle = (TextView) view.findViewById(R.id.item_remark_title);
        mItemRemark = (EditText) view.findViewById(R.id.item_remark);
        LinearLayout mItemRecord2 = (LinearLayout) view.findViewById(R.id.item_record2);
        mScrollView = view.findViewById(R.id.check_content);
        mItemType.setText(contentItem.getCheaktype());
        mItemSection.setText(contentItem.getCheakdatail());
        mItemContent.setText(contentItem.getXuhao() + "、" + contentItem.getCheakname());
        if (!TextUtils.isEmpty(contentItem.getCheakrecord())) {
            mItemRecord1.setVisibility(View.VISIBLE);
            mItemRecordTitle.setText(contentItem.getCheakrecord());
        }
        if (!TextUtils.isEmpty(contentItem.getCheakrecord2())) {
            mItemRecord2.setVisibility(View.VISIBLE);
            mItemRemarkTitle.setText(contentItem.getCheakrecord2());
        }
        if (type) {
            mItemRecord.setEnabled(false);
            mItemRecord.setHint("");
            mItemRemark.setEnabled(false);
            mItemRemark.setHint("");
            mCheckResultRadioFailure.setEnabled(false);
            mCheckResultRadioOk.setEnabled(false);
        } else {
            mCheckResultRadioGroup.setOnCheckedChangeListener(this);
        }
    }

    private void changeTextColor(boolean isOk) {
        if (isOk) {
            mCheckResultRadioOk.setTextColor(getResources().getColor(R.color.colorText));
            mCheckResultRadioFailure.setTextColor(getResources().getColor(R.color.colorSecondDrayText));
        } else {
            mCheckResultRadioOk.setTextColor(getResources().getColor(R.color.colorSecondDrayText));
            mCheckResultRadioFailure.setTextColor(getResources().getColor(R.color.colorText));
        }
    }

    @Override
    public void onCheckedChanged(RadioGroup group, int checkedId) {
        if (checkedId == R.id.check_result_radio_ok) {
            checkResultItem.ischeck = "1";
            changeTextColor(true);
        } else {
            checkResultItem.ischeck = "0";
            changeTextColor(false);
        }
    }


    @Override
    public void onPause() {
        super.onPause();
        saveData();
    }

    public void saveData() {
        checkResultItem.content = mItemRecord.getText().toString();
        checkResultItem.remark = mItemRemark.getText().toString();
        ApplianceCheckResultItem.getBox().put(checkResultItem);
    }

    public boolean verify() {
        return mCheckResultRadioFailure.isChecked() && ((mItemRecord.getVisibility() == View.GONE || mItemRecord.getText().length() > 0) || (mItemRemark.getVisibility() == View.GONE || mItemRemark.getText().length() > 0));
    }

    public void scrollToBottom() {
        mScrollView.fullScroll(ScrollView.FOCUS_DOWN);
    }

}

File: app/src/main/java/com/bkjcb/rqapplication/check/fragment/ApplianceCheckItemDetailFragment.kt
package com.bkjcb.rqapplication.check.fragment

import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.bkjcb.rqapplication.R
import com.bkjcb.rqapplication.check.model.ApplianceCheckContentItem
import com.bkjcb.rqapplication.check.model.ApplianceCheckResultItem
import com.bkjcb.rqapplication.check.model.ApplianceCheckResultItem_
import com.bkjcb.rqapplication.base.datebase.DataUtil
import com.bkjcb.rqapplication.base.fragment.BaseLazyFragment
import kotlinx.android.synthetic.main.fragment_appliance_check_item_detail.*

/**
 * Created by DengShuai on 2020/1/7.
 * Description :
 */
class ApplianceCheckItemDetailFragment : BaseLazyFragment(), RadioGroup.OnCheckedChangeListener {
    private lateinit var contentItem: ApplianceCheckContentItem
    private var uid //项目id
            : String? = null
    private var checkResultItem: ApplianceCheckResultItem? = null
    private var type = false
    fun setType(type: Boolean) {
        this.type = type
    }

    fun setContentItem(contentItem: ApplianceCheckContentItem) {
        this.contentItem = contentItem
    }

    fun setUid(uid: String?) {
        this.uid = uid
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (rootView == null) {
            rootView = inflater.inflate(R.layout.fragment_appliance_check_item_detail, null)
            initView()
        }
        return view
    }

    override fun onStart() {
        super.onStart()
        checkResultItem = queryLocalItem()
        if (checkResultItem != null) {
            item_record.setText(checkResultItem!!.content)
            item_remark.setText(checkResultItem!!.remark)
            if (!TextUtils.isEmpty(checkResultItem!!.ischeck)) {
                if ("1" == checkResultItem!!.ischeck) {
                    check_result_radioGroup.check(R.id.check_result_radio_ok)
                    changeTextColor(true)
                } else if ("0" == checkResultItem!!.ischeck) {
                    check_result_radioGroup.check(R.id.check_result_radio_failure)
                    changeTextColor(false)
                }
            }
        } else {
            checkResultItem = ApplianceCheckResultItem(uid!!, contentItem.guid!!)
        }
    }

    private fun queryLocalItem(): ApplianceCheckResultItem? {
        return DataUtil.getApplianceCheckResultItemBox().query().equal(ApplianceCheckResultItem_.jianchaid, uid!!).and().equal(ApplianceCheckResultItem_.jianchaxiangid, contentItem.guid!!).build().findFirst()
    }

    private fun initView() {
        item_type.text = contentItem.cheaktype
        item_section.text = contentItem.cheakdatail
        item_content.text = "${contentItem.xuhao}、${contentItem.cheakname}"
        if (!TextUtils.isEmpty(contentItem.cheakrecord)) {
            item_record1.visibility = View.VISIBLE
            item_record_title.text = contentItem.cheakrecord
        }
        if (!TextUtils.isEmpty(contentItem.cheakrecord2)) {
            item_record2.visibility = View.VISIBLE
            item_remark_title.text = contentItem.cheakrecord2
        }
        if (type) {
            item_record.isEnabled = false
            item_record.hint = ""
            item_remark.isEnabled = false
            item_remark.hint = ""
            check_result_radio_failure.isEnabled = false
            check_result_radio_ok.isEnabled = false
        } else {
            check_result_radioGroup.setOnCheckedChangeListener(this)
        }
    }

    private fun changeTextColor(isOk: Boolean) {
        if (isOk) {
            check_result_radio_ok.setTextColor(resources.getColor(R.color.colorText))
            check_result_radio_failure.setTextColor(resources.getColor(R.color.colorSecondDrayText))
        } else {
            check_result_radio_ok.setTextColor(resources.getColor(R.color.colorSecondDrayText))
            check_result_radio_failure.setTextColor(resources.getColor(R.color.colorText))
        }
    }

    override fun onCheckedChanged(group: RadioGroup, checkedId: Int) {
        if (checkedId == R.id.check_result_radio_ok) {
            checkResultItem!!.ischeck = "1"
            changeTextColor(true)
        } else {
            checkResultItem!!.ischeck = "0"
            changeTextColor(false)
        }
    }

    override fun onPause() {
        super.onPause()
        saveData()
    }

    fun saveData() {
        checkResultItem!!.content = item_record.text.toString()
        checkResultItem!!.remark = item_remark.text.toString()
        DataUtil.getApplianceCheckResultItemBox().put(checkResultItem!!)
    }

    fun verify(): Boolean {
        return check_result_radio_failure.isChecked && (item_record!!.visibility == View.GONE || item_record.text.isNotEmpty() || item_remark.visibility == View.GONE || item_remark.text.isNotEmpty())
    }

    fun scrollToBottom() {
        check_content.fullScroll(ScrollView.FOCUS_DOWN)
    }

    companion object {
        fun newInstances(contentItem: ApplianceCheckContentItem, id: String?): ApplianceCheckItemDetailFragment {
            val fragment = ApplianceCheckItemDetailFragment()
            fragment.setContentItem(contentItem)
            fragment.setUid(id)
            return fragment
        }

        fun newInstances(contentItem: ApplianceCheckContentItem, id: String?, type: Boolean): ApplianceCheckItemDetailFragment {
            val fragment = ApplianceCheckItemDetailFragment()
            fragment.setContentItem(contentItem)
            fragment.setUid(id)
            fragment.setType(type)
            return fragment
        }
    }
}